### DISCUSSION TITLE
Is Codeigniter version 3.1.10 is support for php 8.0 or php 8.1 ?

### DISCUSSION THREAD

**Discussion by vkdevhero** [bcit-ci/CodeIgniter#6269] [Q&A]
Hello all
I hope you are doing well.

I have one question regarding whether CodeIgniter version 3.1.10 is compatible with PHP 8.0 or 8.1? 
If yes, Can you tell me what changes is needed to upgrade to PHP version 8.0 or 8.1?

Thanks

**Comment by jp26198926** [comment]
Hi @vkdevhero, not sure for 3.1.10 but 3.1.13 is okay with php8.0 and php8.1 and then fail on php8.2 in my machine.

**Comment by vkdevhero** [comment]
@jp26198926 
Ok thanks.

**Comment by mariolucasdev** [comment]
yes, im use currently with php 8.1 and its ok;

**Comment by ganl** [comment]
There seems to be a lot of MR about PHP 8.x compatibility
https://github.com/bcit-ci/CodeIgniter/pulls?q=is%3Apr+is%3Aopen+PHP+8+

@gxgpet

**Comment by LakshmiPrem** [comment]
redirect not working

**Comment by mohsetiawanmodeong** [comment]
in php 8.2.4 can't running and error

**Comment by mariolucasdev** [comment]
@vkdevhero and @mohsetiawanmodeong 

I currently have 3 projects running with php 8.1 and codeigniter 3.1.13 and it is working perfectly.

It doesn't work with 8.2, just as codeigniter with versions prior to 3.1.13 doesn't work with php 8.1.

Updating Codeigniter is very simple, you can see this documentation page.

https://codeigniter.com/userguide3/installation/upgrading.html

**Comment by mohsetiawanmodeong** [comment]
Thanks for the advice,
So how to upgrade from ci 3.1.10 to 3.1.13 directly?
I'm using php 8.2.4, and ci 3.1.10 still error.

Thanks,
Wawan

**Comment by mariolucasdev** [comment]
@mohsetiawanmodeong 

you have to follow the step by step of each version, 3.1.10 > 3.1.11 until you reach the most recent one.

Another thing is that with php 8.2 codeigniter 3.1.13 does not work.

Version 3.1.13 works well with php 8.1.

**Comment by mohsetiawanmodeong** [comment]
Ahh i c,
So, CI 3 not support again with php 8.2.4?
Thanks dude.

**Comment by kenjis** [comment]
No, CI 3.1.13 does not work with PHP 8.2 at all.
You need to upgrade it by yourself.

**Comment by mohsetiawanmodeong** [comment]
oke thanks dude.
can you give me a tutorial for upgrade to PHP 8.2.4?

Pada Sel, 20 Feb 2024 pukul 09.00 kenjis ***@***.***> menulis:

**Comment by kenjis** [comment]
There are no tutorials available to upgrade CodeIgniter 3.1.13 for PHP 8.2.
However, PR #6173 is helpful.

Also, at least there is one fork project for PHP 8.2:
https://github.com/pocketarc/codeigniter
But it has some minor breaking changes.
See https://github.com/pocketarc/codeigniter/issues/3

**Comment by pocketarc** [comment]
It's worth noting that the breaking changes come from the fork being based on the "develop" branch of CI, not from any changes made as part of the fork. These are being reverted so that there are no breaking changes in CI 3.2.

I don't think there should be any breaking changes for CI 3. CI 3 is the legacy version of CI, maintained only to keep old projects alive, so it makes sense to try not to break anything.
